cockpit/src/main/java/com/polidea/cockpit/manager/CockpitManager.kt

```diff
@@ -1,55 +1,78 @@
 package com.polidea.cockpit.manager
 
-import com.polidea.cockpit.core.Param
+import com.polidea.cockpit.core.CockpitParam
 import com.polidea.cockpit.event.PropertyChangeListener
-import kotlin.collections.ArrayList
+import com.polidea.cockpit.utils.FileUtils
+import com.polidea.cockpit.utils.copy
+import com.polidea.cockpit.utils.getParam
 
 object CockpitManager {
 
-    val params: MutableList<Param<*>> = ArrayList()
+    private val params: MutableList<CockpitParam<Any>> by lazy {
+        FileUtils.getParams().toMutableList()
+    }
+
+    val defaultParams: List<CockpitParam<Any>> by lazy {
+        FileUtils.getDefaultParams()
+    }
 
-    var defaultParams: MutableList<Param<*>> = mutableListOf()
+    private val paramChangeNotifier = ParamChangeNotifier()
 
-    fun addParam(param: Param<*>) {
+    fun addParam(param: CockpitParam<*>) {
         checkIfExistsAndAddParam(param)
     }
 
-    private fun checkIfExistsAndAddParam(param: Param<*>) {
+    private fun checkIfExistsAndAddParam(param: CockpitParam<*>) {
         if (!exists(param.name)) {
             System.out.println("Param ${param.name} doesn't exist, adding")
-            params.add(NotifiableParam(param.name, param.value, param.description, param.group))
+            params.add(CockpitParam(param.name, param.value, param.description, param.group))
         } else {
             System.out.println("Param ${param.name} already exists")
         }
     }
 
-    fun <T : Any> getParamValue(name: String): T? = getParam<Param<T>>(name).value
+    private fun exists(key: String) =
+            params.find { it.name == key } != null
 
-    fun <T : Any> setParamValue(key: String, value: T) {
-        val param = getParam<Param<T>>(key)
-        param.value = value
+    fun <T : Any> getParamValue(name: String): T = params.getParam<CockpitParam<T>>(name).value
+
+    fun <T : Any> getParamDefaultValue(name: String): T = defaultParams.getParam<CockpitParam<T>>(name).value
+
+    fun setParamsValue(params: Collection<CockpitParam<Any>>) {
+        params.forEach { setParamValueWithAutoSave(it.name, it.value, false) }
+        save()
     }
 
-    fun <T : Any> addOnParamChangeListener(key: String, listener: PropertyChangeListener<T>) {
-        val param = getParam<NotifiableParam<T>>(key)
-        param.addPropertyChangeListener(listener)
+    fun <T : Any> setParamValue(name: String, newValue: T) {
```

Comment:
Look that setParamValueWithAutoSave is private and called from public methods - setParamValue and setParamsValue. The reason is to prevent user from saving the file manually.

Reply:
Yes, but setParamValue saves the file each time it's called. Which is ok, because its called from autogenerated Cockpit class. setParamValues is used internally and we want save the file after setting all values. There is no need to save it after setting a single param value